src/openapi-registry.ts

```diff
@@ -14,26 +16,39 @@ import {
   SchemaObject,
   SecuritySchemeObject,
 } from 'openapi3-ts';
-import type { ZodVoid, ZodObject, ZodSchema, ZodType } from 'zod';
+import type { ZodObject, ZodSchema, ZodType } from 'zod';
 
 type Method = 'get' | 'post' | 'put' | 'delete' | 'patch';
 
-export type ResponseConfig =
-  // Matching ResponseObject in openapi3-ts
-  | {
-      mediaType: string;
-      schema: ZodType<unknown>;
-      description?: string;
-      headers?: HeadersObject;
-      links?: LinksObject;
-    }
-  | ZodVoid;
+export interface ZodMediaTypeObject {
+  schema: ZodType<unknown>;
```

Comment:
Do we want to allow custom schemas to be passed here instead of only Zod schemas? I'm not sure how response formats which are not json could be described using Zod schemas. For example, XML attributes?